docs/DeveloperGuide.md

```diff
@@ -1,34 +1,513 @@
+
+The `Ui` component
+
+* Reads user input as lines using `readLine()`.
+* Prints every `Command` object that is parsed using `printMessage()`.
+* Prints any user input error that might occur using `showError()`.
+
+![UiSequence](diagrams/UiSequenceDiagram.png)
+
+<small><i>Figure 3</i></small>
+
+### Logic component
+
+![LogicDiagram](diagrams/LogicClassDiagram.png)
+
+<small><i>Figure 4</i></small>
+
+1. `Logic` stores a current `Template` object in [`Model`](#model-component) that represents the current circuit configuration.
+1. `Logic` uses the `Parser` class to parse the user command.
+1. This results in a `Command` object which is executed in `Duke`.
+1. The command execution can affect the `Model` (e.g. setting a value).
+1. In addition, the `Ui` may also perform certain actions, such as displaying help to the user.
+
+### Model component
+
+![ModelDiagram](diagrams/ModelClassDiagram.png)
+
+<small><i>Figure 5</i></small>
+
+The `Model`,
+* includes `CircuitTemplate` and `BooleanTemplate` that can represent the current `template` in [`Logic`](#logic-component).
+* has `Component` and `Gate` within the templates.
+* does not depend on any of the other three components.
+
+--------------------------------------------------------------------------------------------------------------------
+
+## Implementation of Circuit Commands (Dash) <a name="circ-comd"></a>
+
+This section provides details on the implementation of the various electronic circuit commands.
+
+There are 4 different types of components of electronic circuits that can be instantiated in the program:
+* `Resistor` - A _resistor_ component.
+* `Capacitor` - A _capacitor_ component.
+* `Inductor` - An _inductor_ component.
+* `VoltageSource` - An _Alternating Current_ Voltage Source.
+
+Each component is used within a circuit template, based on the kind of circuit instantiated.
+
+<div markdown="span" class="alert alert-primary">
+
+:information_source: **Note:** The `VoltageSource` is instantiated in all Circuit Templates.
+
+</div>
+
+There are four different circuit templates that can be instantiated in the program:
+
+* `LcTemplate` - Inductor-Capacitor Circuit Template
+* `RTemplate` - Resistor Circuit Template
+* `RcTemplate` - Resistor-Capacitor Template (extends `RTemplate`)
+* `LrTemplate` - Inductor-Resistor Template (extends `RTemplate`)
+
+![CircuitCommandClass](diagrams/CircuitCommandClassDiagram.png)
+
+<small><i>Figure 6</i></small>
+
+The diagram above demonstrates the relationship between the various `CircuitCommand` objects. The various commands to be parsed are as explained in this section. While the **User Guide** explains the commands used on the CLI, this section goes into detail the classes used to execute the commands. The command classes also make use of the classes in the [**`Model`**](#model-component) shown in the diagram below when executing the commands.
+
+![CircuitModelClass](diagrams/CircuitModelClassDiagram.png)
+
+<small><i>Figure 7</i></small>
```

Comment:
Maybe this diagram should appear before the `CircuitCommandClassDiagram`, because it immediately follows the textual description of the template structure. Also, would a separate diagram relating the `Template` classes to the `Command` classes be good?

Reply:
Thanks, should be part of https://github.com/AY2021S1-CS2113T-W13-3/tp/pull/121.